src/scramble_parser.rs

```diff
@@ -65,7 +65,41 @@ fn get_variant(mv: &str) -> MoveVariant {
     }
 }
 
-/// Simplifies the given moves by combining adjacent moves.
+/// Recursively merges adjacent moves of the same Move discriminant
+/// until no further simplification is possible.
 pub fn simplify_moves(moves: &[Move]) -> Vec<Move> {
-    vec![]
+    use std::mem::discriminant;
+    let mut result = vec![];
+    if moves.is_empty() {
+        return result;
+    }
+
+    // keep track of the current move and its amount of clockwise turns
+    type Movement = (Move, u8);
+    let mut movement: Movement = (moves[0], moves[0].get_variant() as u8);
+
+    // returns a Move if the simplified movement has any effect on a cube
+    fn movement_to_move(m: Movement) -> Option<Move> {
+        match m.1 % 4 {
+            1 => Some(m.0.with_variant(MoveVariant::Standard)),
+            2 => Some(m.0.with_variant(MoveVariant::Double)),
+            3 => Some(m.0.with_variant(MoveVariant::Inverse)),
+            _ => None,
+        }
+    }
+
+    // merge adjacent moves of the same type
+    for mv in moves[1..].iter() {
+        if discriminant(&movement.0) == discriminant(mv) {
```

Comment:
Following from the previous point. It might be beneficial to create an entirely new ``Move`` are assign it back to ``movement``. This is certainly less efficient, but I think it expresses the intent of the code a little better.